quasar-core/src/main/java/co/paralleluniverse/fibers/instrument/MethodDatabase.java

```diff
@@ -134,8 +133,17 @@ public Log getLog() {
     }
 
     public Set<String> getClassNames() {
-        // Create a "snapshot" of the current set of class names.
-        return new HashSet<>(classes.keySet());
+        synchronized(classes) {
+            // Create a "snapshot" of the current set of class names.
+            return new HashSet<>(classes.keySet());
+        }
+    }
+
+    public Set<String> getClassNamesForSuperClasses() {
+        synchronized(superClasses) {
```

Comment:
Any risk due to now locking classes and superClasses separately e.g. need to hold both locks somewhere and risk deadlock, or non-atomic update?

Reply:
I don't see how, because I only hold a lock to access a `Map`, and no-one needs to access both `Map`s simultaneously.